contracts/contracts/subnet/SubnetActorManagerFacet.sol

```diff
@@ -67,6 +69,11 @@ contract SubnetActorManagerFacet is SubnetActorModifiers, ReentrancyGuard, Pausa
         payable(msg.sender).sendValue(amount);
     }
 
+    function setValidatorGater(address gater) external notKilled {
```

Comment:
The gater needs to be updated this way as compared to in a constructor because the gater needs the subnet address, but the subnet is not created at the constructor. Alternatively, one can also create the gater first then set the subnet address in gater. But I figured this is less intrusive to downstream ipc rust code and in the redesign, we can update accordingly.